### DISCUSSION TITLE
Query an entry by values of a replicator field?

### DISCUSSION THREAD

**Discussion by geertjanknapen1** [statamic/cms#9705] [Q&A]
I'm wondering if I can query an Entry by the values of the replicator field.

I have a collection of products.
It contains a replicator field: Variants
Each of those replicator sets has a variant_id field.

I want to get the Product entry if the Variants Replicator field contains one of X variant ids.

Do you think this can be done or am I pursuing the impossible?

**Comment by jasonvarga** [comment]
You can using json operators.

```php
Entry::query()
->where('replicator_field->0->sub_field', 'value')
->get()
```

See https://github.com/statamic/cms/pull/4758

**Reply by geertjanknapen1** [reply]
This means I have to do it like this right ?

`replicator->0->field`
`replicator->1->field`
`replicator->2->field`
...
...

I guess that works if I can count the amount of items in the replicator. Thanks
